elide-core/src/main/java/com/yahoo/elide/core/pagination/Pagination.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2015, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.core.pagination;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.ToString;
+
+import javax.ws.rs.core.MultivaluedMap;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Encapsulates the pagination strategy.
+ */
+
+@AllArgsConstructor
+@ToString
+public class Pagination {
+
+    public static final int DEFAULT_PAGE_SIZE = 500;
+    // todo do we ever want to return more than 10000 items? Set Upper Bounds
+
+    private static final Pagination DEFAULT_PAGINATION = new Pagination(0, 0);
+
+    public static final Map<String, String> PAGE_KEYS = new HashMap<>();
+    static {
+        PAGE_KEYS.put("page[size]", "pageSize");
```

Comment:
I followed http://jsonapi.org/format/#fetching-pagination for the page fields. All but cursor.